### PR TITLE
Add library crate definition back

### PR DESCRIPTION
**Pull request by Veykril**
https://github.com/ferrocene/specification/pull/511#discussion_r1665544257

### CODE REVIEW

File: src/program-structure-and-compilation.rst [hunk 1]
```diff
@@ -138,6 +138,9 @@ The :t:`function` in scope of a :t:`binary crate`'s :t:`crate root module` under
 the :t:`name` ``main`` with a :t:`main function signature` is the :t:`binary
 crate`'s :t:`program entry point`.
 
+:dp:`fls_d9nn4yuiw1ja`
+A :t:`library crate` is a :t:`crate` that is not a :t:`binary crate`.
```
Comment (pietroalbini) [inline src/program-structure-and-compilation.rst:142]: This is not correct, as a proc macro crate is also not a binary crate. I'd specify that it's a crate with the type of `lib`, `rlib`, `staticlib`, `dylib` or `cdylib`.

**Review by pietroalbini — CHANGES_REQUESTED** [review]

File: src/program-structure-and-compilation.rst (hunk 1 above)
Comment (Veykril) [reply src/program-structure-and-compilation.rst:142]: But that definition says so, a proc-macro crate is not a binary crate so it fits the definition of a library crate. Anyways you are right that we ought to be explicit here by listing them.

File: src/changelog.rst [hunk 2]
```diff
@@ -74,7 +74,6 @@ Language changes in Rust 1.79.0
     * Updated glossary entries: :t:`binary crate`, :t:`proc-macro crate`
     * Updated paragraphs: :p:`fls_9ub6ks8qrang`, :p:`fls_Mf62VqAhoZ3c`
```
Comment (pietroalbini) [inline src/changelog.rst:75]: We also updated glossary entries and paragraphs.

File: src/changelog.rst [hunk 3]
```diff
@@ -72,9 +72,8 @@ Language changes in Rust 1.79.0
     * New glossary entry: :t:`crate type`
     * New paragraphs: :p:`fls_unxalgMqIr3v`, :p:`fls_e7jGvXvTsFpC`, :p:`fls_kQiJPwb2Hjcc`, :p:`fls_OyFwBtDGVimT`
     * Updated glossary entries: :t:`binary crate`, :t:`proc-macro crate`
```
Comment (pietroalbini) [inline src/changelog.rst:74]:
```suggestion
    * Updated glossary entries: :t:`binary crate`, :t:`library crate`, :t:`proc-macro crate`
```

File: src/changelog.rst [hunk 4]
```diff
@@ -72,9 +72,8 @@ Language changes in Rust 1.79.0
...
-    * Updated paragraphs: :p:`fls_9ub6ks8qrang`, :p:`fls_Mf62VqAhoZ3c`
+    * Updated paragraphs: :p:`fls_9ub6ks8qrang`, :p:`fls_Mf62VqAhoZ3c`, :p:`fls_d9nn4yuiw1ja`, :p:`fls_3m8lg4mdc2x0`
```
Comment (pietroalbini) [inline src/changelog.rst:75]:
```suggestion
    * Updated paragraphs: :p:`fls_9ub6ks8qrang`, :p:`fls_Mf62VqAhoZ3c`, :p:`fls_d9nn4yuiw1ja`
```

**Review by pietroalbini — APPROVED** [review]

**Review by pietroalbini — APPROVED** [review]
bors merge

**Pull request merged**